cardinal/world.go

```diff
@@ -429,20 +432,21 @@ func (w *World) startGameLoop(ctx context.Context, tickStart <-chan time.Time, t
 	}()
 }
 
-func (w *World) tickTheEngine(ctx context.Context, tickDone chan<- uint64) {
+func (w *World) tickTheEngine(ctx context.Context, tickDone chan types.TickResult) {
 	currTick := w.CurrentTick()
 	// this is the final point where errors bubble up and hit a panic. There are other places where this occurs
 	// but this is the highest terminal point.
 	// the panic may point you to here, (or the tick function) but the real stack trace is in the error message.
-	if err := w.Tick(ctx, uint64(time.Now().Unix())); err != nil {
+	err := w.ticks(ctx, uint64(time.Now().Unix()))
+	if err != nil {
 		bytes, err := json.Marshal(eris.ToJSON(err, true))
 		if err != nil {
 			panic(err)
 		}
 		w.Logger.Panic().Err(err).Str("tickError", "Error running Tick in Game Loop.").RawJSON("error", bytes)
 	}
 	if tickDone != nil {
-		tickDone <- currTick
+		tickDone <- types.TickResult{CurrentTick: currTick, Error: err}
```

Comment:
Since a non-nil error on line 441 always results in a panic, this Error field will always be nil.

We should either 1) reserve the tickDone channel for successfully completed ticks (reverting this channel's type back to a uint64 instead of a struct) or 2) Ensure errors actually make it out of this function by putting this if statement between lines 440 and 441. 

My weakly held opinion is option 1.

Reply:
ohh, i think panic should be only called when `json.Marshal` returning error, so the code should be like this :
```
err := w.doTick(ctx, uint64(time.Now().Unix()))
if err != nil {
	bytes, errMarshal := json.Marshal(eris.ToJSON(err, true))
	if errMarshal != nil {
		panic(errMarshal)
	}
	w.Logger.Panic().Err(err).Str("tickError", "Error running Tick in Game Loop.").RawJSON("error", bytes)
}
```

The return of error in line 449 now is used by `DoTick` func in `TestFixture`, it's on `cardinal/testutils/world.go`

Reply:
Per our conversation: We DO want w.Logger.Panic()... to panic, so Msg needs to be attached to that line.

This way, any tick errors that are encountered will results in a panic, causing cardinal to shut down.

the tickDone channel doesn't need an error field in this case.

The test in tick_test.go "TestIfPanicMessageLogged" becomes a little more complicated since it no longer has access to the public Tick function. I suggest reverting this test back into its original code, and moving it to a new file called `world_test.go`, with a package of `cardinal`. This will give you access to the private doTick function.

In general we should try to put our tests in `<package>_test` to force us to only test public facing APIs, but this situations seems like an appropriate place to deviate from that rule.